python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -650,84 +577,208 @@ async def do_setPowerManagementMode(self, data: salobj.BaseMsgType) -> None:
             mode=self.mtdome_com.power_management_mode
         )
 
-    async def statusAMCS(self) -> None:
-        """AMCS status command not to be executed by SAL.
+    async def status_amcs(self, status: dict[str, typing.Any]) -> None:
+        """AMCS status command.
 
-        This command will be used to request the full status of the AMCS lower
-        level component.
+        Parameters
+        ----------
+        status : `dict`[`str`, `typing.Any`]
+            The status.
         """
-        await self.request_and_send_llc_status(LlcName.AMCS, self.tel_azimuth)
+        if "exception" in status:
+            self.log.error(status["exception"])
+            await self.go_fault(self.status_amcs)
```

Comment:
You could abstract this part for the other functions to reuse.

Reply:
I could but I am lazy ;)